test/automation/src/sql/notebook.ts

```diff
@@ -117,6 +117,32 @@ export class Notebook {
 		await this.code.waitForElementGone(Notebook.placeholderSelector);
 	}
 
+	async waitForCollapseIconInCells(): Promise<void> {
+		let cellIds = await this.getCellIds();
+		for (let i of cellIds) {
+			const editor = `.notebook-cell[id="${i}"] code-cell-component code-component collapse-component`;
+			await this.code.waitForElement(`${editor} [title="Collapse code cell contents"]`);
+		}
+	}
+
+	async waitForExpandIconInCells(): Promise<void> {
+		let cellIds = await this.getCellIds();
+		for (let i of cellIds) {
+			const editor = `.notebook-cell[id="${i}"] code-cell-component code-component collapse-component`;
+			await this.code.waitForElement(`${editor} [title="Expand code cell contents"]`);
+		}
+	}
+
+	// Helper function to get cell ids for notebook
+	async getCellIds(): Promise<string[]> {
+		let cellIds: string[] = [];
+		await this.code.waitForElements('div.notebook-cell', false, result => {
+			cellIds = result.map(cell => cell.attributes['id']);
```

Comment:
The elements are returned from this so you don't have to use the callback - you can just map the results directly